### DISCUSSION TITLE
How to disable language modules at once

### DISCUSSION THREAD

**Discussion by noraworld** [starship/starship#3145] [Q&A] [answered]
I would like to disable all language modules to enhance performance on my shell prompt of Starship.

I know I can do this, but I think I’m too lazy to disable all language modules one by one as follow.

```
[ruby]
disabled = true

[python]
disabled = true

[php]
disabled = true

[nodejs]
disabled = true

# ...
```

How can I do what I want to more ellegant way?

Thank you.

**Comment by matchai** [comment, marked as answer]
If you only want a select few modules to be enabled, this would best be done by setting the root-level `format` value.

For instance, the following would give you a minimal, directory-only prompt:
```toml
format = """
$directory\
$newline\
$character
"""
```

**Comment by noraworld** [comment]
@matchai This will help me do what I want! Thank you for letting me know :pray:

On the other hand, I’m still curious about disabling specific modules by group like language modules. Can’t this be done at the moment?

**Reply by matchai** [reply]
Not currently. There's no notion of groupings in Starship at the moment.

**Reply by noraworld** [reply]
I see! Thank you very much!

I hope this feature will be implemented in the future 😉
